var/spack/repos/builtin/packages/wi4mpi/package.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Wi4mpi(CMakePackage):
+    """WI4MPI: Wrapper Interface For MPI performing a light translation between MPI
+    constants and MPI objects from an MPI implementation to another one"""
+
+    homepage = "https://github.com/cea-hpc/wi4mpi"
+    url      = "https://github.com/cea-hpc/wi4mpi/archive/v3.4.1.tar.gz"
+    maintainers = ['adrien-cotte', 'marcjoos-cea']
+
+    version('3.4.1', sha256='92bf6738216426069bc07bff19cd7c933e33e397a941ff9f89a639380fab3737')
+    version('3.3.0', sha256='fb7fb3b591144e90b3d688cf844c2246eb185f54e1da6baef857e035ef730d96')
+    version('3.2.2', sha256='23ac69740577d66a68ddd5360670f0a344e3c47a5d146033c63a67e54e56c66f')
+    version('3.2.1', sha256='0d928cb930b6cb1ae648eca241db59812ee0e5c041faf2f57728bbb6ee4e36df')
+    version('3.2.0', sha256='3322f6823dbec1d58a1fcf163b2bcdd7b9cd75dc6c7f78865fc6cb0a91bf6f94')
+    variant('build_type', default='Release',
+            description='The build type to build',
+            values=('Debug', 'Release', 'RelWithDebInfo'))
+
+    depends_on('mpi')
+
+    def cmake_args(self):
+        if '%gcc' in self.spec:
+            compiler = "GNU"
+        elif '%intel' in self.spec:
+            compiler = "INTEL"
+        elif '%clang' in self.spec:
+            compiler = "LLVM"
+        elif '%pgi' in self.spec:
+            compiler = "PGI"
+        else:
+            tty.error("Could not determine compiler used")
+        wi4mpi_build_type = 'RELEASE'
+        if self.spec.variants["build_type"].value == "RelWithDebInfo":
+            wi4mpi_build_type = 'NORMAL'
+        elif self.spec.variants["build_type"].value == "Debug":
+            wi4mpi_build_type = 'DEBUG'
+        args = ["-DWI4MPI_REALEASE=" + wi4mpi_build_type,

```

Comment:
Maybe a typo:
```suggestion
        args = ["-DWI4MPI_RELEASE=" + wi4mpi_build_type,

```
?

Reply:
Also note that a more idiomatic way to write this is:
```python
args = [
    self.define('WI4MPI_RELEASE', wi4mpi_build_type),
    self.define('WI4MPI_COMPILER', compiler)
]
```

Reply:
It is indeed a typo but in Wi4MPI base code itself. We will fix it in a future release and change it accordingly in Spack recipe.

Thank you for the suggestion to improve argument passing; I've just pushed a new commit with the suggested edit.